pypesto/result/optimize.py

```diff
@@ -223,3 +232,18 @@ def get_for_key(self, key) -> list:
             "releases."
         )
         return [res[key] for res in self.list]
+
+    def _assign_unique_id(self, id: str, suffix: int = None):
+        """
+        Assign a unique id to an id if needed.
+
+        Returns id if not used already otherwise tries id_suffix with
+        increasing suffix until it is unique.
+        """
+        if suffix is None:
+            if id not in self.id:
```

Comment:
Where is `self.id` defined?

Reply:
in recent changes it was changed so that now `self.id` is equal to `self.get_for_key(id)`

Reply:
Could you link me to where it's defined? Can't find it in this class.

Reply:
https://github.com/ICB-DCM/pyPESTO/blob/4d737237ed85dfb9f051af6a1331e85666f62f24/pypesto/result/optimize.py#L145

Reply:
Thanks!